Share.cpp

```diff
@@ -10,39 +10,40 @@ void Share::assertMatches()
 {
 	list<Individual*>::iterator i , ii;
 	Match * m;
-
 	for ( i = matches.begin() ; i != matches.end() ; i++ )
 	{
 		if (DEBUG) cout << "i in share::assertMatches outer loop: " << *i << endl;
 		ii = i;
 		for ( ++ii ; ii != matches.end() ; ii++ )
 		{
-			if (DEBUG) cout << "ii in share::assertMatches inner loop: " << *ii << endl;
-			// Check if this pair matched in previous word (symmetrically)
-			m = (*i)->getMatch( (*ii)->getNumericID() );
-			if (DEBUG) cout << "numericID in share::assertMatches inner loop: " << (*ii)->getNumericID() << endl;
-			if ( m == NULL ) m = (*ii)->getMatch( (*i)->getNumericID() );
-
-			if ( m != NULL )
-			{
-				// This match can be incremented
-				if (DEBUG) cout << "incrementing match end_ms to " << position_ms << endl;
-				m->end_ms = position_ms;
-			}
-			else
-			{
-				// This match must be created
-				m = createMatch( *i , *ii );
-				// Extend the match backwards
-				if (DEBUG) cout << "extendBack() called from Share.cpp" << endl;
-				m->extendBack();
-				// Mark asserted
-				(*i)->addMatch( (*ii)->getNumericID() , m );
+			if ((!ALL_SAMPLES.hasRestrictions()) || (ALL_SAMPLES.isOld((*i)->getBaseID()) && ALL_SAMPLES.isNew((*ii)->getBaseID())) || (ALL_SAMPLES.isNew((*i)->getBaseID()) && ALL_SAMPLES.isOld((*ii)->getBaseID()))) {
```

Comment:
Ok, so it looks like we're only comparing "old" to "new", so if you want to compare 10 dogs to themselves-and-1000-other-dogs, you'd have a 10-line new.txt and a 1010-line old.txt